js_modules/dagster-ui/packages/ui-core/src/hooks/useStableReferenceByHash.tsx

```diff
@@ -7,8 +7,20 @@ import {hashObject} from '../util/hashObject';
  * It is useful to avoid unsubscribing/re-subscribing to the same keys in case the reference changes but the keys are the same.
  * It is also useful to avoid re-rendering the component when the value changes but the hash is the same.
  */
-export const useStableReferenceByHash = <T,>(value: T, hashFn = hashObject) => {
+
+const referenceCache = new Map<string, any>();
+
+export const useStableReferenceByHash = <T,>(value: T, storeInMap = false, hashFn = hashObject) => {
```

Comment:
I think it'd be helpful to elaborate on when you should pass true vs false here. I assume that storeInMap should only be used when the value T is low-variance (eg: flipping between three values) and it should not be used when the range of values is huge or T is constantly changing?

Reply:
storeInMap is when you want the value to be cached in memory globally. This is if you expect that other instances of the hook might run into the same value and if it it useful for performance globally to keep that reference the same. The use case I'm using it for now is for caching the object describing asset statuses. The reason I want to cache the reference globally is because (1) I believe other instances of the hook will run into the same statuses and (2) we refresh status information every 45 seconds and if the status information hasn't changed then I want to be able to preserve the references. Let me see if I can come up with a succinct comment